server/src/api/user.rs

```diff
@@ -439,6 +479,56 @@ impl Perform for Oper<Register> {
   }
 }
 
+#[async_trait::async_trait(?Send)]
+impl Perform for Oper<GetCaptcha> {
+  type Response = GetCaptchaResponse;
+
+  async fn perform(
+    &self,
+    _pool: &DbPool,
+    websocket_info: Option<WebsocketInfo>,
+  ) -> Result<Self::Response, LemmyError> {
+    let captcha_settings = Settings::get().captcha;
+
+    if !captcha_settings.enabled {
+      return Ok(GetCaptchaResponse {
+        png: "disabled".to_string(),
+        uuid: "disabled".to_string(),
+        wav: None,
```

Comment:
This is very hacky, should return an error message instead.

Reply:
I'll think of another way to do this. My option are either giving an error field, or just wrapping the whole thing in an option.